### PR TITLE
updated parser

### LINKED ISSUE #4998: update parser
some issues in the parser will benefit codegen users.  See https://github.com/swagger-api/swagger-parser/pull/412

### PR DESCRIPTION
**Pull request by fehguy**
Fixes #4998

### CODE REVIEW

**Comment by ePaul** [comment]
Hmm, This seems to be not liked by any CI.
Travis says:

```
Tests run: 632, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 13.555 sec <<< FAILURE! - in TestSuite
petstoreOperationTest(io.swagger.codegen.bash.BashTest)  Time elapsed: 0.031 sec  <<< FAILURE!
java.lang.ClassCastException: java.util.LinkedHashMap cannot be cast to com.fasterxml.jackson.databind.node.ObjectNode
	at io.swagger.codegen.bash.BashTest.petstoreOperationTest(BashTest.java:42)
```

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/BashClientCodegen.java [hunk 1]
```diff
@@ -571,16 +573,17 @@ public CodegenOperation fromOperation(String path, String httpMethod,
         List codesamples = (List)op.vendorExtensions.get("x-code-samples");
 
         for (Object codesample : codesamples) {
-          ObjectNode codesample_object = (ObjectNode)codesample;
+            if(codesample instanceof ObjectNode) {
+                ObjectNode codesample_object = (ObjectNode) codesample;
 
-          if((codesample_object.get("lang").asText()).equals("Shell")) {
+                if ((codesample_object.get("lang").asText()).equals("Shell")) {
```
Comment (fehguy) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/BashClientCodegen.java]: TBH I have no idea what this code is supposed to do, but I do know it's an unsafe type cast.  So... I'm making it safe, but still don't know why this is here.

File: modules/swagger-codegen/src/test/java/io/swagger/codegen/bash/BashTest.java [hunk 2]
```diff
@@ -48,7 +32,7 @@ public void petstoreOperationTest() {
                 swagger);
 
         Assert.assertTrue(
-            op.vendorExtensions.containsKey("x-bash-codegen-sample"));
+            op.vendorExtensions.containsKey("x-code-samples"));
```
Comment (fehguy) [inline modules/swagger-codegen/src/test/java/io/swagger/codegen/bash/BashTest.java]: The key name didn't match what is in the spec...

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/BashClientCodegen.java (hunk 1 above)
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/BashClientCodegen.java]: cc @bkryza

**Pull request merged**